### PR TITLE
Automanage SDK track 2 generation and tests

### PR DESCRIPTION
**Pull request by The-DevBlog**
- Track 2 SDK generation
- Tests to cover all main operations of the SDK

### CODE REVIEW

File: sdk/automanage/Azure.ResourceManager.Automanage/src/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+
+[assembly: Azure.Core.AzureResourceProviderNamespace("automanage")]
```
Comment (ArthurMa1978) [inline sdk/automanage/Azure.ResourceManager.Automanage/src/Properties/AssemblyInfo.cs:6]: Microsoft.Automanage

File: sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
+# Generated code configuration
+Run `dotnet build /t:GenerateCode` to generate code.
+``` yaml
+save-inputs: true
+azure-arm: true
+csharp: true
+library-name: automanage
+namespace: Azure.ResourceManager.automanage
```
Comment (ArthurMa1978) [inline sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md:8]: Upper case the first character of `automanage`

File: sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+# require: https://github.com/Azure/azure-rest-api-specs/blob/d32cece9ca8814ef42085d4bbc426dc35bbcaf87/specification/automanage/resource-manager/readme.md
+require: https://github.com/AndrewCS149/azure-rest-api-specs/blob/360b7c3331b919bfe90148adff6abe025d21c720/specification/automanage/resource-manager/readme.md
```
Comment (ArthurMa1978) [inline sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md:10]: The sdk should use a file from [github.com/Azure/azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs), not personal repo

File: sdk/automanage/ci.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  branches:
+    include:
+    - main
+    - feature/*
+    - hotfix/*
+    - release/*
+  paths:
+    include:
+    - sdk/automanage/
+    - sdk/resourcemanager/
```
Comment (ArthurMa1978) [inline sdk/automanage/ci.yml:23]: remove this line

File: sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md (hunk 3 above)
Comment (AndrewCS149) [reply sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md:10]: Oops! I meant to switch it. Done!

File: sdk/automanage/Azure.ResourceManager.Automanage/src/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (AndrewCS149) [reply sdk/automanage/Azure.ResourceManager.Automanage/src/Properties/AssemblyInfo.cs:6]: Done

File: sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md (hunk 2 above)
Comment (AndrewCS149) [reply sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md:8]: Done

File: sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+  Ipv6: IPv6|ipv6
+  Ipsec: IPsec|ipsec
+  SSO: Sso
+  URI: Uri
+  Etag: ETag|etag
+directive:
+#use scope parameter on the two paths that are defined multiple times
+  - from: automanage.json
+    where: $.paths
+    transform: $['/{scope}/providers/Microsoft.Automanage/configurationProfileAssignments/{configurationProfileAssignmentName}'] = $['/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/providers/Microsoft.Automanage/configurationProfileAssignments/{configurationProfileAssignmentName}']
```
[the remark this answers is not in the document]
Comment (AndrewCS149) [reply sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md:51]: Sounds good. @m-nash created this autorest.md to help get the SDK generated.

File: sdk/automanage/ci.yml (hunk 4 above)
Comment (AndrewCS149) [reply sdk/automanage/ci.yml:23]: Done

File: sdk/automanage/test-resources.bicep [hunk 6]
```diff
@@ -0,0 +1,84 @@
+@minLength(12)
```
Comment (benbp) [inline sdk/automanage/test-resources.bicep:1]: I'm happy to see the bicep usage! Please just commit the bicep file and not the ARM template, as the latter is unnecessary and will result in two deployments (which will resolve idempotently to the same set of resources, but will take longer to run).

Comment (AndrewCS149) [reply sdk/automanage/test-resources.bicep:1]: Thanks for reviewing @benbp! I'm actually using the ARM template to create some Virtual machine within some of the test cases. Is it fine if I just rename the ARM template?

Comment (benbp) [reply sdk/automanage/test-resources.bicep:1]: Ah I see. Yes it should be fine as long as you rename it so our automation won't pick it up by convention.

File: sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/BestPracticeResource.cs [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+namespace Azure.ResourceManager.Automanage
+{
+    /// <summary>
+    /// A Class representing a BestPractice along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct a <see cref="BestPracticeResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetBestPracticeResource method.
+    /// Otherwise you can get one from its parent resource <see cref="TenantResource" /> using the GetBestPractice method.
+    /// </summary>
+    public partial class BestPracticeResource : ArmResource
```
Comment (m-nash) [inline sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/BestPracticeResource.cs:26]: Each resource should have CRUD tests associated with it

File: sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
+# Generated code configuration
+Run `dotnet build /t:GenerateCode` to generate code.
+``` yaml
+save-inputs: true
```
Comment (m-nash) [inline sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md:4]: remove this line so the CodeModel files go away

File: sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/ConfigurationProfileVersionResource.cs [hunk 9]
```diff
@@ -0,0 +1,430 @@
...
+
+namespace Azure.ResourceManager.Automanage
+{
+    /// <summary>
+    /// A Class representing a ConfigurationProfileVersion along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct a <see cref="ConfigurationProfileVersionResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetConfigurationProfileVersionResource method.
+    /// Otherwise you can get one from its parent resource <see cref="ConfigurationProfileResource" /> using the GetConfigurationProfileVersion method.
+    /// </summary>
+    public partial class ConfigurationProfileVersionResource : ArmResource
```
Comment (m-nash) [inline sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/ConfigurationProfileVersionResource.cs:26]: each resource should have CRUD tests associated with it

File: sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/BestPracticeVersionResource.cs [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+
+namespace Azure.ResourceManager.Automanage
+{
+    /// <summary>
+    /// A Class representing a BestPracticeVersion along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct a <see cref="BestPracticeVersionResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetBestPracticeVersionResource method.
+    /// Otherwise you can get one from its parent resource <see cref="BestPracticeResource" /> using the GetBestPracticeVersion method.
+    /// </summary>
+    public partial class BestPracticeVersionResource : ArmResource
```
Comment (m-nash) [inline sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/BestPracticeVersionResource.cs:25]: Each resource should have CRUD tests associated with it

File: sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/Extensions/AutomanageExtensions.cs [hunk 11]
```diff
@@ -0,0 +1,446 @@
...
+            {
+                return new ArmResourceExtensionClient(client, armResource.Id);
+            }
+            );
+        }
+
+        /// <summary> Gets a collection of ConfigurationProfileAssignmentResources in the ArmResource. </summary>
+        /// <param name="armResource"> The <see cref="ArmResource" /> instance the method will execute against. </param>
+        /// <returns> An object representing collection of ConfigurationProfileAssignmentResources and their operations over a ConfigurationProfileAssignmentResource. </returns>
+        public static ConfigurationProfileAssignmentCollection GetConfigurationProfileAssignments(this ArmResource armResource)
```
Comment (m-nash) [inline sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/Extensions/AutomanageExtensions.cs:295]: Are we 100% sure this method applies to all resources?

There were some changes to how this is generated and your branch is using an [old version of autorest](https://github.com/AndrewCS149/azure-sdk-for-net/blob/andrsmith/automanageSDKTests/eng/Packages.Data.props#L163).  Sync from main and regenerate.  The default is now to take the id of the resource target as an extension off of ArmClient.  If this truly should apply to ALL resources then there is a configuration to generate it as it shows now which is an extension off of ArmResource.

**Comment by m-nash** [comment]
Also missing the API / snippets run the following from root of your repo
`.\eng\scripts\Export-API.ps1 automanage`
`.\eng\scripts\Update-Snippets.ps1 automanage`

File: sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/ConfigurationProfileVersionResource.cs (hunk 9 above)
Comment (AndrewCS149) [reply sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/ConfigurationProfileVersionResource.cs]: Used directives to exclude some of the operations that are not quite customer ready.

File: sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/BestPracticeVersionResource.cs (hunk 10 above)
Comment (AndrewCS149) [reply sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/BestPracticeVersionResource.cs]: Used directives to exclude some of the operations that are not quite customer ready.

File: sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md (hunk 8 above)
Comment (AndrewCS149) [reply sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md:4]: @m-nash do you mean line 4?

File: sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/Extensions/AutomanageExtensions.cs (hunk 11 above)
Comment (AndrewCS149) [reply sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/Extensions/AutomanageExtensions.cs]: Thanks for pointing this out. Ive re-generated the SDK and modified the tests to support this.

File: sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md (hunk 8 above)
Comment (AndrewCS149) [reply sdk/automanage/Azure.ResourceManager.Automanage/src/autorest.md:4]: Done

**Comment by AndrewCS149** [comment]
#sign-off

File: sdk/textanalytics/Azure.AI.TextAnalytics/README.md [hunk 12]
```diff
@@ -252,7 +252,7 @@ Please refer to the service documentation for a conceptual discussion of [sentim
 Run a model to identify a collection of significant phrases found in the passed-in document or batch of documents.
 
 ```C# Snippet:ExtractKeyPhrases
-string document = @"My cat might need to see a veterinarian. It has been sneezing more than normal, and although my
+string document = @"My cat might need to see a veterinarian. It has been sneezing more than normal, and although my
```
Comment (m-nash) [inline sdk/textanalytics/Azure.AI.TextAnalytics/README.md:255]: can we remove this?

File: sdk/automanage/ci.yml [hunk 13]
```diff
@@ -0,0 +1,28 @@
+# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
+
+trigger:
```
Comment (m-nash) [inline sdk/automanage/ci.yml:3]: rerun eng/scripts/Update-Mgmt-CI.ps1 to update this to the new format

File: sdk/textanalytics/Azure.AI.TextAnalytics/README.md (hunk 12 above)
Comment (AndrewCS149) [reply sdk/textanalytics/Azure.AI.TextAnalytics/README.md:255]: done

File: sdk/automanage/ci.yml (hunk 13 above)
Comment (AndrewCS149) [reply sdk/automanage/ci.yml]: done

**Comment by m-nash** [comment]
/azp prepare net - automanage - mgmt - ci

**Review by m-nash — APPROVED** [review]
For GA we will want to address the naming of BestPractice and ConfigurationProfile.  These two objects are technically the same thing except one is built in and the other is custom.  They should be named to reflect this relationship to make it easier for customers.

**Pull request merged**